pkg/cluster/memberlist.go

```diff
@@ -0,0 +1,54 @@
+package cluster
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net"
+	"strconv"
+	"time"
+
+	"github.com/hashicorp/memberlist"
+)
+
+type delegate interface {
+	memberlist.EventDelegate
+	memberlist.Delegate
+}
+
+func NewMemberlist(name, addr string, secret []byte, verbose bool, d delegate) (*memberlist.Memberlist, error) {
+	if len(secret) != 32 {
+		return nil, fmt.Errorf("invalid secret size, must be 32 bytes: %d", len(secret))
+	}
+
+	host, portString, err := net.SplitHostPort(addr)
+	if err != nil {
+		return nil, fmt.Errorf("address must be a host:port: %v", err)
+	}
+
+	port, err := strconv.Atoi(portString)
+	if err != nil {
+		return nil, fmt.Errorf("address must be a host:port: %v", err)
```

Comment:
copy-pasta: the error message here should not be about `host:port` but about `port` not being an integer.

Reply:
:spaghetti: :+1: